Sources/Sentry/SentryUIViewControllerPerformanceTracker.m

```diff
@@ -109,6 +120,53 @@ - (void)createTransaction:(UIViewController *)controller
     }
 }
 
+- (void)createTimeToDisplay:(UIViewController *)controller
+{
+    SentrySpan *vcSpan = [self viewControllerPerformanceSpan:controller];
+
+    if (![vcSpan isKindOfClass:[SentryTracer self]]) {
+        // Since TTID and TTFD are meant to the whole screen
+        // we will not track child view controllers
+        return;
+    }
+
+    if ([vcSpan.tracer getExtensionsOfType:SentryTimeToDisplayTracker.self].count > 0) {
+        // Already tracking time to display, not creating a new tracker.
+        // This may happen if user manually call `loadView` from a view controller more than once.
+        return;
+    }
+
+    if (self.currentTTDTracker) {
+        [self.currentTTDTracker stopWaitingFullDisplay];
+    }
+
+    SentryTimeToDisplayTracker *ttdTracker =
+        [[SentryTimeToDisplayTracker alloc] initForController:controller
+                                           waitForFullDisplay:self.enableWaitForFullDisplay];
+    [vcSpan.tracer addExtension:ttdTracker];
+
+    self.currentTTDTracker = ttdTracker;
+}
+
+- (void)finishTTIDForController:(UIViewController *)controller
+{
+    SentrySpan *vcSpan = [self viewControllerPerformanceSpan:controller];
+
+    if (![vcSpan isKindOfClass:[SentryTracer self]]) {
+        // Only tracer may have TimeToDisplayTracker
+        return;
+    }
+
+    SentryTimeToDisplayTracker *ttdTracker =
+        [vcSpan.tracer getExtensionsOfType:SentryTimeToDisplayTracker.self].firstObject;
+    [ttdTracker reportInitialDisplay];
+}
+
+- (void)reportFullyDisplayed
+{
+    [self.currentTTDTracker reportFullDisplay];
```

Comment:
Please standardize the names. Either both should be `reportFullyDisplayed` or `reportFullDisplay`